src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+                               login_server,
+                               username,
+                               password,
+                               image)
+        raise CLIError("{} Please check if image/tag was inputted correctly.".format(str(e)))
+
+# Validate if blob layer can be pulled
+def _check_blob(repository_name,
+                login_server,
+                username,
+                password,
+                image):
+
+    import requests
+    from ._docker_utils import get_manifest_authorization_header, parse_error_message
+
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+
+    manifest_response = requests.get(
+        manifest_url,
+        headers=get_manifest_authorization_header(username, password))
+
+    manifest = manifest_response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+    digest = smallest_blob['digest']
+
+    # Get blob
+    request_url = _get_blob(login_server, repository_name, digest)
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+
+    res = requests.get(request_url, auth=(username, password))
+
+    # Return server error message; if server error not available, will return error message below
+    if res.status_code >= 400:
+        raise CLIError(parse_error_message('Could not get the requested data.', res), res.status_code)
+    # Check if image or tag was inputted and return appropriate message
+    if res.status_code < 400 and 'sha256' not in image:
+        print_pass("Blobs referenced by tag '{}' can be pulled from registry '{}'".format(image, login_server))
+    else:
+        print_pass("Blobs in image '{}' can be pulled from registry '{}'".format(image, login_server))
 
 # General command
 def acr_check_health(cmd,  # pylint: disable useless-return
                      vnet=None,
                      ignore_errors=False,
                      yes=False,
-                     registry_name=None):
+                     registry_name=None,
+                     repository_name=None,
+                     image=None,
+                     resource_group_name=None):
     from azure.cli.core.util import in_cloud_console
+    from .repository import get_access_credentials
+    from ._utils import get_resource_group_name_by_registry_name
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, resource_group_name)
+
+    registry = None
+
+    # Connectivity
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name, rg)
+        login_server = registry.login_server.rstrip('/')
+    except CLIError:
+        from ._docker_utils import get_login_server_suffix
+        suffix = get_login_server_suffix(cmd.cli_ctx)
+
+        if not suffix:
```

Comment:
why? add comment please

Reply:
This edit was added in response to above comment by Matt where he pointed out that get_registry_by_name and get_access_credentials were being used multiple times in different methods. So I am calling it once the acr_check_health and passing the info into the subfunctions. I will add comments so this is clear.

Reply:
I meant add a comment explaining what are you validation with the "if not suffix" it's not clear at first glance by looking at the code

Reply:
Sure, sounds good, will add comment here. This particular block of code was already existing, I just moved its location. Agree that there is a lack of explanation, thank you!